includes/mailpoet/class-sensei-mailpoet.php

```diff
@@ -0,0 +1,382 @@
+<?php
+/**
+ * File containing the class Sensei_MailPoet.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( \MailPoet\API\API::class ) ) {
+	return;
+}
+
+/**
+ * MailPoet integration class.
+ *
+ * Handles the integration with the MailPoet plugin,
+ * creates a list for each course and group, adds enrolled students.
+ *
+ * @package Core
+ * @since $$next-version$$
+ */
+class Sensei_MailPoet {
+
+	/**
+	 * MailPoet API handle.
+	 *
+	 * @var object
+	 */
+	private $mailpoet_api;
+
+	/**
+	 * Instance of the current handler.
+	 *
+	 * @var Sensei_MailPoet
+	 */
+	private static $instance;
+
+	/**
+	 * A list of Sensei Courses and Groups.
+	 *
+	 * @var array
+	 */
+	private $sensei_lists;
+
+	/**
+	 * All lists on MailPoet.
+	 *
+	 * @var array
+	 */
+	private $mail_poet_lists;
+
+	/**
+	 * Constructor
+	 *
+	 * @since $$next-version$$
+	 * @param object $mailpoet_api MailPoet API handle.
+	 */
+	public function __construct( $mailpoet_api ) {
+		$this->mailpoet_api = $mailpoet_api;
+		if ( $this->mailpoet_api->isSetupComplete() ) {
```

Comment:
I'm trying to test the PR. At the moment I'm getting this error when running the background job:
```
[21-Mar-2023 19:21:09 UTC] PHP Fatal error:  Uncaught Error: Call to a member function isSetupComplete() on array in /Users/merkushin/Developer/sensei/includes/mailpoet/class-sensei-mailpoet.php:63
Stack trace:
#0 /Users/merkushin/Developer/sensei/includes/mailpoet/class-sensei-mailpoet.php(85): Sensei_MailPoet->__construct(Array)
#1 /Users/merkushin/Developer/sensei/includes/mailpoet/class-sensei-mailpoet-sync-job.php(53): Sensei_MailPoet::instance(Array)
#2 /Users/merkushin/Developer/sensei/includes/background-jobs/class-sensei-background-job-batch.php(47): Sensei_MailPoet_Sync_Job->run_batch(0)
#3 /Users/merkushin/Developer/sensei/includes/background-jobs/class-sensei-scheduler-action-scheduler.php(63): Sensei_Background_Job_Batch->run()
#4 /Users/merkushin/Developer/sensei/includes/background-jobs/class-sensei-scheduler.php(72): Sensei_Scheduler_Action_Scheduler->run(Object(Sensei_MailPoet_Sync_Job), Object(Closure))
#5 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(308): Sensei_Scheduler::run_stateful_job(Array)
#6 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#7 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/plugin.php(565): WP_Hook->do_action(Array)
#8 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/actions/ActionScheduler_Action.php(43): do_action_ref_array('sensei_backgrou...', Array)
#9 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_QueueRunner.php(65): ActionScheduler_Action->execute()
#10 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_ListTable.php(557): ActionScheduler_Abstract_QueueRunner->process_action('6104', 'Admin List Tabl...')
#11 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_ListTable.php(526): ActionScheduler_ListTable->process_row_action('6104', 'run')
#12 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_ListTable.php(615): ActionScheduler_ListTable->row_action_run('6104')
#13 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/abstracts/ActionScheduler_Abstract_ListTable.php(728): ActionScheduler_Abstract_ListTable->process_row_actions()
#14 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_AdminView.php(107): ActionScheduler_Abstract_ListTable->process_actions()
#15 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/packages/action-scheduler/classes/ActionScheduler_AdminView.php(95): ActionScheduler_AdminView->get_list_table()
#16 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(308): ActionScheduler_AdminView->render_admin_ui('')
#17 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#18 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#19 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/includes/admin/views/html-admin-page-status.php(41): do_action('woocommerce_adm...')
#20 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-status.php(23): include_once('/Users/merkushi...')
#21 /Users/merkushin/Local Sites/devwp/app/public/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-menus.php(305): WC_Admin_Status::output()
#22 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(308): WC_Admin_Menus->status_page('')
#23 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#24 /Users/merkushin/Local Sites/devwp/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#25 /Users/merkushin/Local Sites/devwp/app/public/wp-admin/admin.php(259): do_action('woocommerce_pag...')
#26 {main}
  thrown in /Users/merkushin/Developer/sensei/includes/mailpoet/class-sensei-mailpoet.php on line 63
```

MailPoet is installed and set up.
<img width="865" alt="CleanShot 2023-03-22 at 02 22 17@2x" src="https://user-images.githubusercontent.com/329356/226719379-05504368-e3a3-466f-bb86-a70b600dc7c9.png">
<img width="990" alt="CleanShot 2023-03-22 at 02 25 03@2x" src="https://user-images.githubusercontent.com/329356/226719394-c4fc1924-ff7c-4541-a7ae-b83606e15033.png">

I'm looking for a problem in my env, but maybe you know the cause?

Reply:
Another problem is that I had ~30 pages of pending jobs for the last 30 minutes 😱